src/units/animation.cpp

```diff
@@ -1345,6 +1346,29 @@ void unit_animator::add_animation(unit_const_ptr animated_unit
 	animated_units_.push_back(std::move(tmp));
 }
 
+bool unit_animator::has_animation(unit_const_ptr animated_unit
+		, const std::string& event
+		, const map_location &src
+		, const map_location &dst
+		, const int value
+		, const unit_animation::hit_type hit_type
+		, const_attack_ptr attack
+		, const_attack_ptr second_attack
+		, int value2) const
+{
+	if(!animated_unit){
+		return false;
+	}
+
+	display* disp = display::get_singleton();
+
+	if(animated_unit->anim_comp().choose_animation(*disp, src, event, dst, value, hit_type, attack, second_attack, value2)){
+		return true;
+	}
+
+	return false;
+}
```

Comment:
I think this can be even more simply written as:
```
return animated_unit
	&& animated_unit->anim_comp().choose_animation(*display::get_singleton(), src, event, dst, value, hit_type, attack, second_attack, value2);
```

Reply:
it is done